### ISSUE TITLE
How to crop video clip in Flowblade? FAQ is incorrect.

### ISSUE THREAD

**Issue by luv** [jliljebl/flowblade-forum#43]
Hi

Im trying to crop a clip in flowblade and the FAQ (from Jan 2018) says "Affine in filter group Transform and set properties **Scale X, Scale Y**, X, Y."

https://github.com/jliljebl/flowblade/blob/master/flowblade-trunk/docs/FAQ.md#crop-filter-does-not-work

I'm trying this in latest Flowblade installable from .deb (2.2.0, Aug 2019) but there is no Scale X or Scale Y property in that filter.

How can i crop (zoom) a video clip in flowblade?

**Comment by luv** [comment]
For some clips transform->translate works but for other clips it does not and it moves the clip in the preview up/down instead of changing size/height of the clip.

**Comment by jliljebl** [comment]
Yes, for 2.2 we did some changes in **'Transform'** filters group.

The correct filter to use for this usecase is  now **'Position Scale**'.

Unfortunately we have a bug: the numerical inputs do not work, but the GUI editor does, so use that.

**Comment by luv** [comment]
Thanks for your answer. So zoom works correctly only in 2.4 which can't be easy installed using .deb package, correct? (I can't use flatpak as it unfortunately has just too many security and integration issues.)

**Comment by jliljebl** [comment]
I had an error in the answer above, the changes were made for 2.4 that came out a month ago. There was still one bug with the numerical inputs. Zoom will work correctly and with a GUI editor in 2.6 coming June.

**Comment by luv** [comment]
Thank you. You have done really good job on flowblade. When I tried it recently I was really surprised that there is finally a free easy to use linux video editor which is not crashing randomly etc. I will try again when 2.6 is out.

**Comment by luv** [comment]
In the meantime I will use flowblade 2.2 with ffmpeg/ffplay for cropping:
```

        ffplay -i input.mp4 -vf crop=100:100:50:10
        ffmpeg -i input.mp4 -vf crop=w:h:x:y output.mp4

```

**Comment by luv** [comment]
@jliljebl Actually I dug into the code, it's pretty easy to follow :+1: and I figured out what's wrong, mlt (at least 6.6.0 does *not* ignore the width parameter if the parameter is different that the original clip width thus why it works for some video but not for others - the resize works only for videos that have same width as the profile (the comment here is not 100% correct https://github.com/jliljebl/flowblade/blob/master/flowblade-trunk/Flowblade/propertyedit.py#L522 ).

I forked flowblade and created a new branch "branch_v2.2" branched from v2.2 tag - I pushed one commit to this branch scaling the width using the same factor as used for height.

Here is the simple fix: https://github.com/luv/flowblade/commit/3c74624c3544a9437be7dc094d18ed9aa57898f1 

I understand 2.2 is no longer developed and this functionality is completely reworked in 2.4 and going forward, however I think it'd be great to fix this in 2.2 as it's the current .deb release and flowblade is in my opinion the best simple video editor on linux yet cropping is pretty basic functionality.

**Comment by jliljebl** [comment]
> @jliljebl Actually I dug into the code, it's pretty easy to follow +1 and I figured out what's wrong,..  mlt (at least 6.6.0 does not ignore the width parameter if the parameter is different that the original clip width thus why it works for some video but not for others...

Alright, damn, I wish I had got that right. I have to say that this is also some bad API on MLTs part, functionality should not depend on if clip and profile have same width, couldn't guess that there was more to this issue then what seemed at first glance.

Unfortunately I cannot bring myself to burn any energy on fixing and old .deb, . The position is "use Flatpak or repo code"

**Comment by luv** [comment]
@jliljebl I completely understand, packaging is pain and there is so many packages now that a project should support to make everyone happy (debs, rpms, appimage, snap, flatpak).

I have used your script from flowblade-packaging, modified it a little bit and copied 'control' file from the latest released deb (2.2-1) and ran your script against my "branch_v2.2" and changed version to 2.2-2.

The built .deb is here: https://github.com/luv/flowblade/releases/tag/v2.2-2

You can verify with 

```
debdiff original/flowblade-2.2.0-1_all.deb patched/flowblade-2.2.0-2_all.deb
```

and

```
mkdir original
cd original
ar x flowblade-2.2.0-1_all.deb 
mkdir flowblade_unpacked
tar xf control.tar.xz -C flowblade_unpacked/
tar xf data.tar.xz -C flowblade_unpacked/
cd ..
mkdir patched
cd patched
ar x flowblade-2.2.0-2_all.deb
mkdir flowblade_unpacked/
tar xf data.tar.xz -C flowblade_unpacked/
tar xf control.tar.gz -C flowblade_unpacked/
cd ..
diff -r original/flowblade_unpacked/ patched/flowblade_unpacked/
```

Only changes are the version and the 3 lines of code. Hope others find this useful.

Thanks again for working on flowblade.

**Comment by luv** [comment]
Btw. you might also consider dropping flowblade-packaging for building .deb packages and fork debian's https://salsa.debian.org/multimedia-team/flowblade instead and make the changes as necessary (change package's maintainer in control etc.).

All I did to build .deb using the above repo:
```
apt install git-buildpackage
# change  "pristine-tar=False" and "sign-tags = False" in debian/gbp.conf
apt install build dependencies as show by the buildpackage on first run
gbp buildpackage -uc -ui -us
```

**Comment by jliljebl** [comment]
@luv Thanks on the pointer on Debian packaging resource, my own build script is probably a bit...uh...let's say idiosyncratic. I don't think anyone else does .debs like that. I used to use some Debian tools tool for packaging but somehow doing my own script seemed easier. I give a go on doing package the Debian way next time and see how it goes.

I'll put a link to your package on the installing page, now that you have taken the time to create this (thanks!) I think users should be aware of it existing.
